Firestore/core/src/firebase/firestore/immutable/sorted_map.h

```diff
@@ -161,6 +161,16 @@ class SortedMap : public SortedMapBase {
     UNREACHABLE();
   }
 
+  const C& comparator() const {
```

Comment:
Is this an intentional change?

Reply:
Yes, though not yet for this PR.

In [FSTView](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Core/FSTView.mm#L249) the bit that compares two documents is actually creating a new comparator for each comparison. Changing that line to reuse the comparator from the DocumentSet avoids that needless work. This change enables that.

Admittedly this isn't strictly related to this PR but I lumped it in here because there was a vague relationship.